addons/website_sale_stock/static/src/js/website_sale_stock.js

```diff
@@ -38,12 +38,15 @@ $('.oe_website_sale').each(function() {
     $(oe_website_sale).on('change', 'ul[data-attribute_value_ids]', function(event) {
         var $ul = $(event.target).closest('.js_add_cart_variants');
         var $parent = $ul.closest('.js_product');
-        var variant_ids = JSON.parse($ul.data("attribute_value_ids").replace(/'/g, '"'));
+        var variant_ids = $ul.data("attribute_value_ids");
+        if(_.isString(variant_ids)) {
+        	variant_ids = JSON.parse(variant_ids.replace(/'/g, '"'));
```

Comment:
attribute_value_ids ==     
def get_attribute_value_ids(self, product):
        """ list of selectable attributes of a product """

So I don't see why we should have a boolean or smthing other than list serialized here

Reply:
Well, it is cheap to check for a string to not have JS errors in case something else is found, right? Why shouldn't we allow it? Not sure if I need this for my inherited stuff as I turn off the event and overwrite it anyway, but honestly...this QWeb compatibility monkey patch is hell...

Reply:
It is about ```website_sale_stock``` which actually adds a dict ;-)